### PR TITLE
docs(user): size the exp doctest tolerance to the operator's own accuracy

### PR DESCRIPTION
**Pull request by lyfne123**
## Summary

Three runnable blocks in the user manual compare a device `pl.exp` against `torch.exp` and
asserted `rtol=1e-4, atol=1e-4` — the bound the surrounding elementwise blocks use. `exp`
turns the device's relative error into a proportionally larger absolute one, and on the CI
simulator runners that error reaches ~1.5e-4, so the assertion sat below the operator's own
accuracy. The two-task block in `01-task-granularity.md` failed on 3 of the 14 most recent
`main` runs of `docs-examples`, always on the same `two_tasks_via_gm` assertion, missing on
12–29 of 16384 elements with a greatest relative difference of ~1.47e-4 against the 1e-4
limit. After this change those three comparisons are sized to the operator rather than to
the reference, and a reader copying the example no longer inherits a tolerance that measures
`exp` instead of the transformation being taught.

Only `rtol` moves. `atol` stays at `1e-4`, because it is what the bound rests on for the
small outputs — there the device's relative error is a negligible absolute one, so raising it
would buy slack no failure asks for and would let a real regression through on exactly the
elements the relative bound does not protect.

No kernel, pass, or API behaviour changes — the examples themselves are untouched, and the
elementwise blocks keep `1e-4`.

## Changes

- `docs/{en,zh}/user/performance/01-task-granularity.md`: both assertions of the merge-InCore
  block go to `rtol=1e-3` (`atol` unchanged), plus a short paragraph saying why this block is looser
  than the elementwise ones above it.
- `docs/{en,zh}/user/performance/02-runtime-overhead.md`: the `early_resolve` block carries
  the same producer→GM→`exp` shape and the same latent failure; same `rtol`.
- `docs/{en,zh}/user/performance/04-incore.md`: the shared `check()` helper of the two
  double-buffering blocks compares against `torch.exp(A)`; same `rtol`.

The zh pages mirror the en code byte for byte, which `run_doc_examples.py --check-parity`
enforces.

## Verification

- `python tests/docs/run_doc_examples.py --pages docs/en/user/performance --check-parity -p a2a3sim`: exit 0 — `parity OK`, 14/14 runnable doc blocks passed.
- `python tests/lint/check_docs_en_zh_parity.py`: exit 0 — 176 paired markdown paths.
- `python tests/lint/check_docs_nav.py`: exit 0 — 176 nav entries cover all 176 pages.
- `python tests/lint/check_docs_symbol_coverage.py`: exit 0 — 238/238 `pl.__all__` symbols documented.
- `python tests/lint/check_english_only.py`: exit 0 — 1336 files passed.

### CODE REVIEW

File: docs/en/user/performance/04-incore.md [hunk 1]
```diff
@@ -25,7 +25,7 @@ A = (indices % 3 - 1).to(torch.float32).reshape(ROWS, TC)
 def check(kernel):
     out = torch.zeros(ROWS, TC, dtype=torch.float32)
     kernel(A, out, config=CFG)
-    torch.testing.assert_close(out, torch.exp(A), rtol=1e-4, atol=1e-4)
+    torch.testing.assert_close(out, torch.exp(A), rtol=1e-3, atol=1e-3)
```
[the remark this answers is not in the document]
Comment (lyfne123) [reply docs/en/user/performance/04-incore.md]: Good catch — fixed in `c43149f`: only `rtol` moves now, `atol` stays at `1e-4`.

You are right that `atol=1e-3` was the loose end. At `exp(-1)=0.368` it made the threshold `1e-3 + 1e-3*0.368 = 1.37e-3`, i.e. `3.7e-3` relative — 25x the operator error, and looser than the `1e-4` the block started from. With `rtol=1e-3, atol=1e-4` the failure condition becomes `delta > 1e-3 + 1e-4/e`, so the observed device error of ~1.5e-4 clears it by ~6.7x at every magnitude, while the small outputs are held exactly as tightly as before this PR.

Re-ran `python tests/docs/run_doc_examples.py --pages docs/en/user/performance --check-parity -p a2a3sim`: parity OK, 14/14 blocks pass.

**Review by Hzfengsy — APPROVED** [review]
